pil-analyzer/src/condenser.rs

```diff
@@ -939,6 +941,78 @@ fn shift_local_var_refs(e: &mut Expression, shift: u64) {
         .for_each(|e| shift_local_var_refs(e, shift));
 }
 
+fn try_algebraic_expression_to_expression<T: FieldElement>(
+    e: &AlgebraicExpression<T>,
+) -> Result<Expression, EvalError> {
+    Ok(match e {
+        AlgebraicExpression::Reference(AlgebraicReference {
+            name,
+            poly_id: _,
+            next: false,
```

Comment:
Wouldn't be too difficult to implement next:

```
let e = Expression::Reference{...}; // what we currently have
if next {
  UnaryOperation{ op: UnaryOperator::Next, expr: Box::new(e)}.into()
} else {
  e
}
```

Reply:
Not sure where to put this, also, isn't the next already assigned as false?

Reply:
If you remove the `: false`, it matches both fals and true values for next: https://doc.rust-lang.org/reference/patterns.html#destructuring